staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -156,10 +161,9 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			// rule is empty
 			continue
 		}
-		if compiled.TransitionRule {
+		if compiled.TransitionRule && oldObj == nil {
```

Comment:
make sure this nil comparison (and the one above) do the right thing when we have a literal null value for a nullable schema item stored in `oldObj interface{}` (e.g. an `interface{}` variable that is not nil but which is holding a typed nil value)

Reply:
Hm, can we support that? The null value in CEL has type `null_type` (https://github.com/google/cel-spec/blob/master/doc/langdef.md#booleans-and-null), so users may have problems writing expressions that handle an `oldSelf` that may be bound to either `null_type` or some other type.

@jpbetz How are CEL rules on nullable schema items handled today? If we can support them, this would be another bit of KEP text to update:

> The identifier oldSelf is guaranteed to be bound to a non-null value during expression evaluation.

Reply:
Yeah, we might need to update the KEP to make this super clear.

The code matches my expectations for the current behavior, which is that if oldSelf is nil, that the transition rule is not executed.  

I'm in the process of adding tests to cover all the nil cases. 

Does anyone know if I can assume JSON `[]`will always convert to go `[]<type>{}`and not `nil`? If not I'll double check. It makes sense to me that it would (so that there is a distinction between an absent optional array field and a prevent but empty array field).

Reply:
yeah, JSON `[]` → `[]interface{}{}`, not nil